File: linguagens_de_programacao/C/verificar_primo.c
#include <stdio.h>

int main()
{
    int num, aux, div_cont;
    
    printf("digite um número para verificar se é primo: ");
    scanf("%d", &num);
    
    div_cont = 0;
    
    for (aux = num; aux > 0; aux--)
        if (num%aux == 0) {
            div_cont+=1;
        }
        
    if (div_cont != 2) {
        printf("o numero %d não é primo", num);
    }
    else {
        printf("o numero %d é primo", num);
    }
    return 0;
}

File: linguagens_de_programacao/go/verificar_primo.go
package main

import (
	"fmt"
)

func main() {
	var n int
	fmt.Print("digite um número inteiro positivo: ")
	fmt.Scan(&n)

	if n < 2 {
		fmt.Println("não é primo.")
		return
	}

	primo := true
	for i := 2; i*i <= n; i++ {
		if n%i == 0 {
			primo = false
			break
		}
	}

	if primo {
		fmt.Println("é primo.")
	} else {
		fmt.Println("não é primo.")
	}
}
